GhostNetwork.Gateway.Api/Users/UsersController.cs

```diff
@@ -71,12 +71,23 @@ public UsersController(IUsersStorage usersStorage, ICurrentUserProvider currentU
         }
 
         [HttpPut("{userId}/profile-picture")]
+        [RequestSizeLimit(1048576)]
         [ProducesResponseType(StatusCodes.Status204NoContent)]
         public async Task<ActionResult> UpsertProfilePictureAsync(
             IFormFile file,
             [FromRoute] Guid userId,
             CancellationToken cancellationToken)
         {
+            if (userId.ToString() != currentUserProvider.UserId)
+            {
+                return Forbid();
+            }
+
+            if (file.FileName[^4..] != ".jpg" && file.FileName[^4..] != ".png")
```

Comment:
Have you tested this code with file names less than 4 chars?
There is more native way of extracting file extension 
https://docs.microsoft.com/en-us/dotnet/api/system.io.path.getextension?view=net-6.0